File: test-suite/src/test/java/io/micronaut/guice/doc/examples/bindings/imported/ImportedClassesTest.java
package io.micronaut.guice.doc.examples.bindings.imported;

import static org.junit.jupiter.api.Assertions.assertInstanceOf;

import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
import jakarta.inject.Inject;
import org.junit.jupiter.api.Test;

@MicronautTest(startApplication = false)
class ImportedClassesTest {
    @Inject CreditCardProcessor processor;

    @Test
    void testImportedClassIsBound() {
        assertInstanceOf(PayPalCreditCardProcessor.class, processor);
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/guice/doc/examples/bindings/imported/ImportedClassesTest.kt
package io.micronaut.guice.doc.examples.bindings.imported

import io.micronaut.test.extensions.junit5.annotation.MicronautTest
import jakarta.inject.Inject
import org.junit.jupiter.api.Assertions.assertInstanceOf
import org.junit.jupiter.api.Test

@MicronautTest(startApplication = false)
class ImportedClassesTest {
    @Inject lateinit var processor: CreditCardProcessor

    @Test
    fun testImportedClassIsBound() {
        assertInstanceOf(PayPalCreditCardProcessor::class.java, processor)
    }
}
